Module `ReportCharts` not found in Report Creator (Mac OSX)

<!--- This template provides sections for bugs and features. Please delete any irrelevant sections before submitting -->

#### Expected Behavior
<!--- Required. Describe the behavior you expect to see for your case. -->
Be able to use the report creator on all platforms
#### Actual Behavior
<!--- Required. Describe the actual behavior for your case. -->
Report creator stalls half-way because the `ReportCharts` module can not be found on Mac OSX.

```
20200210 16:59:37.960 Trace:: QuantConnect.Report.Report(): Processing live orders
20200210 16:59:37.962 Trace:: PythonInitializer.Initialize(): start...
20200210 16:59:38.926 Trace:: PythonInitializer.Initialize(): ended
[ERROR] FATAL UNHANDLED EXCEPTION: Python.Runtime.PythonException: ModuleNotFoundError : No module named 'ReportCharts'
  at Python.Runtime.Runtime.CheckExceptionOccurred () [0x0000c] in <e353d414d86a43249b2fb6d3c21cfcaa>:0
  at Python.Runtime.PythonEngine.ImportModule (System.String name) [0x00006] in <e353d414d86a43249b2fb6d3c21cfcaa>:0
  at QuantConnect.Report.ReportElements.ChartReportElement..ctor () [0x00015] in <72eeac908cb64a95b5f74b2b37ba8344>:0
  at QuantConnect.Report.ReportElements.MonthlyReturnsReportElement..ctor (System.String name, System.String key, QuantConnect.Packets.BacktestResult backtest, QuantConnect.Packets.LiveResult live) [0x00000] in <72eeac908cb64a95b5f74b2b37ba8344>:0
  at QuantConnect.Report.Report..ctor (System.String name, System.String description, System.String version, QuantConnect.Packets.BacktestResult backtest, QuantConnect.Packets.LiveResult live) [0x001f6] in <72eeac908cb64a95b5f74b2b37ba8344>:0
  at QuantConnect.Report.Program.Main (System.String[] args) [0x00100] in <72eeac908cb64a95b5f74b2b37ba8344>:0
```
#### Potential Solution
<!--- Optional. Describe any potential solutions and/or thoughts as to what may be causing the difference between expected and actual behavior. -->
Investigate why the module is not found
#### Reproducing the Problem
<!--- Required for Bugs. Describe how to reproduce the problem. This can be via a failing unit test or a simplified algorithm that reliably demonstrates this issue.  -->
Run Report creator on Mac OSX with python 3.6.8 (anaconda)
#### System Information
<!--- Required for Bugs. Include any system specific information, such as OS. -->
```
Python 3.6.8 :: Anaconda, Inc.
Python.Runtime.dll.config pointing to: /Users/.../anaconda3/lib/libpython3.6m.dylib
Darwin Kernel Version 1X.2.0
```
#### Checklist
<!--- Confirm that you've provided all the required information. -->
<!--- Required fields --->
- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)
<!--- Required for Bugs, feature request can delete the line below. -->
- [x] I have provided detailed steps to reproduce the issue

<!--- Template inspired by https://github.com/stevemao/github-issue-templates -->

--- a/Common/Python/PythonInitializer.cs
+++ b/Common/Python/PythonInitializer.cs
@@ -14,7 +14,9 @@
  *
 */
 
+using System;
 using System.Collections.Generic;
+using System.IO;
 using System.Linq;
 using Python.Runtime;
 using QuantConnect.Logging;
@@ -60,6 +62,39 @@ namespace QuantConnect.Python
                     PythonEngine.Exec($"import sys;{code}");
                 }
             }
+        }
+
+        /// <summary>
+        /// Adds the provided paths to the end of the PYTHONPATH environment variable, as well
+        /// as the current working directory.
+        /// </summary>
+        /// <param name="extraDirectories">Additional paths to add to the end of PYTHONPATH</param>
+        public static void SetPythonPathEnvironmentVariable(IEnumerable<string> extraDirectories = null)
+        {
+            // create new python path environment variable containing directories
+            var pythonDirectories = new List<string>();
+
+            // Don't include an empty environment variable in pythonPath, otherwise the PYTHONPATH
+            // environment variable won't be used in the module import process
+            var pythonPathEnvironmentVariable = Environment.GetEnvironmentVariable("PYTHONPATH");
+            if (!string.IsNullOrEmpty(pythonPathEnvironmentVariable))
+            {
+                pythonDirectories.Add(pythonPathEnvironmentVariable);
+            }
+
+            // Since the order of the PYTHONPATH matters, let's add any new
+            // entries to the end of the new environment variable to prevent
+            // any potential Python standard library paths from being de-prioritized.
+            if (extraDirectories != null)
+            {
+                pythonDirectories.AddRange(extraDirectories);
+            }
+
+            // Add current directory too, allows python algorithms to find Lean's pre-defined submodules
+            pythonDirectories.Add(new DirectoryInfo(Environment.CurrentDirectory).FullName);
+            var finalPath = string.Join(OS.IsLinux ? ":" : ";", pythonDirectories.Distinct());
+
+            Environment.SetEnvironmentVariable("PYTHONPATH", finalPath);
         }
     }
 }

--- a/Queues/JobQueue.cs
+++ b/Queues/JobQueue.cs
@@ -17,6 +17,7 @@ using System;
 using System.Collections.Generic;
 using System.IO;
 using Newtonsoft.Json;
+using QuantConnect.Python;
 using QuantConnect.Configuration;
 using QuantConnect.Interfaces;
 using QuantConnect.Logging;
@@ -93,21 +94,11 @@ namespace QuantConnect.Queues
 
             if ((Language)Enum.Parse(typeof(Language), Config.Get("algorithm-language")) == Language.Python)
             {
-                // Set the python path for loading python algorithms.
+                // Set the python path for loading python algorithms ("algorithm-location" config parameter)
                 var pythonFile = new FileInfo(location);
-                var pythonPath = new List<string>
-                {
-                    pythonFile.Directory.FullName,
-                    new DirectoryInfo(Environment.CurrentDirectory).FullName,
-                };
-                // Don't include an empty environment variable in pythonPath, otherwise the PYTHONPATH
-                // environment variable won't be used in the module import process
-                var pythonPathEnvironmentVariable = Environment.GetEnvironmentVariable("PYTHONPATH");
-                if (!string.IsNullOrEmpty(pythonPathEnvironmentVariable))
-                {
-                    pythonPath.Add(pythonPathEnvironmentVariable);
-                }
-                Environment.SetEnvironmentVariable("PYTHONPATH", string.Join(OS.IsLinux ? ":" : ";", pythonPath));
+
+                // PythonInitializer automatically adds the current working directory for us
+                PythonInitializer.SetPythonPathEnvironmentVariable(new string[] { pythonFile.Directory.FullName });
             }
 
             //If this isn't a backtesting mode/request, attempt a live job.

--- a/Report/Program.cs
+++ b/Report/Program.cs
@@ -19,6 +19,7 @@ using QuantConnect.Configuration;
 using QuantConnect.Logging;
 using QuantConnect.Orders;
 using QuantConnect.Packets;
+using QuantConnect.Python;
 
 namespace QuantConnect.Report
 {
@@ -29,6 +30,9 @@ namespace QuantConnect.Report
     {
         static void Main(string[] args)
         {
+            // Adds the current working directory to the PYTHONPATH env var.
+            PythonInitializer.SetPythonPathEnvironmentVariable();
+
             // Parse report arguments and merge with config to use in report creator:
             if (args.Length > 0)
             {

